homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -96,3 +112,47 @@ def _async_process_data(self):
             self._icon = FLAME_ICON if self._is_on else IDLE_ICON
 
         self.async_write_ha_state()
+
+
+class PwNotifySensor(PwBinarySensor, BinarySensorEntity):
+    """Representation of a Plugwise Notification binary_sensor."""
+
+    def __init__(self, hass, api, coordinator, name, dev_id, binary_sensor, model):
+        """Set up the Plugwise API."""
+        super().__init__(api, coordinator, name, dev_id, binary_sensor, model)
+
+        self._binary_sensor = binary_sensor
+        self._hass = hass
+
+        self._is_on = False
+        self._icon = None
+        self._attributes = {}
+
+        self._unique_id = f"{dev_id}-{binary_sensor}"
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @callback
+    def _async_process_data(self):
+        """Update the entity."""
+        self._attributes = {}
+
+        notify = self._api.notifications
+
+        self._is_on = False
+        self._state = STATE_OFF
```

Comment:
The binary sensor entity should not inherit from the sensor entity. They have different interfaces. We should just set `is_on` property in the binary sensor. The `state` property should not be overridden.

Make a separate entity base class that doesn't have any non common properties that all platforms can use.

Reply:
Done in 5a5d32f